config.go

```diff
@@ -35,11 +39,33 @@ func defaultSocketPath() string {
 	return filepath.Join(home, ".ssh", "wsl2-ssh-agent.sock")
 }
 
+func where() string {
+	home, err := os.UserHomeDir()
+	if err != nil {
+		log.Fatal(err)
+	}
+	return filepath.Join(home, ".ssh", "wsl2-ssh-agent.sock")
+}
+
+func commandPath(command string) string {
+	path := os.Getenv("PATH")
+	defer os.Setenv("PATH", path)
+	os.Setenv("PATH", path + ":/mnt/c/Windows/System32/WindowsPowerShell/v1.0:/mnt/c/Windows/System32/OpenSSH")
+	path, err := exec.LookPath(command)
+	if err != nil {
+		return ""
+	}
+	return path
+}
+
 func newConfig() *config {
 	c := &config{}
 
 	flag.StringVar(&c.socketPath, "socket", defaultSocketPath(), "a path of UNIX domain socket to listen")
+	flag.StringVar(&c.sshExePath, "ssh-exe", commandPath("ssh.exe"), "a path of Windows ssh.exe")
+	flag.StringVar(&c.powershellPath, "powershell", commandPath("powershell.exe"), "a path of Windows PowerShell (powershell.exe)")
 	flag.BoolVar(&c.foreground, "foreground", false, "run in foreground mode")
+	flag.BoolVar(&c.rawForward, "raw-forward", false, "all messages will be forwarded, including OpenSSH 8.9+ extension messages.")
```

Comment:
Do you want this option? Is it because you don't want `ssh.exe` to be invoked?
In future, I would like to remove the check of `ssh.exe` version after the default OpenSSH on Windows goes to 8.9 or later. So I don't want to include this option so much now.
As a counterproposal, how about assuming 8.9+ if `ssh.exe` is not found or explicitly disabled (by such as `--ssh-path=no`)?

Reply:
Regarding the issue of checking versions, I have a new proposal. I noticed that the details property of ssh-agent for Windows contains the exact version number, I guess we don't need to know the ssh.exe path, just let PowerShell find the ssh-agent and then get its version number.

```powershell
$sshAgent = (Get-Command -CommandType Application ssh-agent.exe -ErrorAction Stop)[0]

Write-Host $sshAgent.Version
```

Reply:
how do you feel?

```go
	ignoreOpenSSHExtensions := !c.rawForward
	if !c.rawForward {
		// check if ssh-agent.exe is older than 8.9
		buffer, err := exec.Command(c.powershellPath, "-Command", "Write-Host (Get-Command -CommandType Application ssh-agent.exe -ErrorAction Stop)[0].Version").Output()
		if err == nil {
			sshAgentVersions := strings.Split(string(buffer), ".")
			log.Printf("ssh-agent.exe version: %s", string(buffer))
			v1, _ := strconv.Atoi(sshAgentVersions[0])
			v2, _ := strconv.Atoi(sshAgentVersions[1])
			ignoreOpenSSHExtensions := v1 <= 8 && v2 < 9
			if ignoreOpenSSHExtensions {
				log.Printf("ssh-agent.exe seems to be old; ignore OpenSSH extension messages")
			}
		} else {
			log.Printf("Not getting ssh-agent.exe version; ignore OpenSSH extension messages")
		}
	}
```

Reply:
Hmm, do you know of any agents other than the old Windows OpenSSH Agent that do not support the extended packet? If needed, I'd like to pass all packets by default, and filter them out only when `--remove-extended-packets` or something is given. I really don't like the filter hack, and don't want to make it permanent by default in the future.

The idea of starting ssh.exe via PowerShell.exe sounds cool, but I concern that it will be slow. Unfortunately, invoking PowerShell.exe is painfully slow. I wanted to make wsl2-ssh-agent wait until repeater.ps1 is ready, but I gave up because my .bashrc took three seconds or so on the first boot.
If we do something, I wonder if we should move both the version check and the filter algorithm into the side of `repeater.ps1`.

Reply:
> do you know of any agents other than the old Windows OpenSSH Agent that do not support the extended packet?

There should be nothing else, OpenSSH 8.9 was released almost 2 years ago, any modern ssh-agent should support it.

Regarding the issue of PowerShell taking too long to execute, moving it to repeater.ps1 is indeed a solution. I will try to write a new repeater.ps1 and look for ways to shorten the waiting time of the Shell.